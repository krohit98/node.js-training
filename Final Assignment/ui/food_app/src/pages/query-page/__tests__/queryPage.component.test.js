import { render, screen } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import QueryPage from '../queryPage.component';
import QueryForm from '../../../components/query-form/queryForm.component';
import routeData from 'react-router';

jest.mock("../../../components/query-form/queryForm.component");
jest.mock("../../../components/query-acknowledgement/querySuccess.component");

describe("Unit testing QueryPage component",()=>{

    beforeEach(()=>{
        jest.spyOn(routeData, 'useParams').mockReturnValue({name:"restaurant name"});
    })

    test("Snapshot testing",()=>{
        const renderer = new ShallowRenderer();
        renderer.render(<QueryPage />);
        const result = renderer.getRenderOutput();

        expect(result).toMatchSnapshot();
    })

    test("Should render QueryPage correctly",()=>{
        QueryForm.mockImplementation(()=><div>QueryFormComponent</div>)
        render(<QueryPage />);
        const heading = screen.getByText("restaurant name");
        const mainText = screen.getByText("Ask us what's on your mind.");
        const queryForm = screen.getByText("QueryFormComponent");
        expect(heading).toBeInTheDocument();
        expect(mainText).toBeInTheDocument();
        expect(queryForm).toBeInTheDocument();
    })
}) 