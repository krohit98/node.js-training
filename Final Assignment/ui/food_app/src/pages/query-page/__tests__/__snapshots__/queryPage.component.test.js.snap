// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing QueryPage component Snapshot testing 1`] = `
<div
  className="pageDiv"
>
  <h1
    className="display-3 pageHeading"
  >
    restaurant name
  </h1>
  <h3
    className="display-6"
  >
    Ask us what's on your mind.
  </h3>
  <QueryForm
    submitHandler={[Function]}
  />
  
  
</div>
`;
